app/models/service_template/copy.rb

```diff
@@ -0,0 +1,17 @@
+module ServiceTemplate::Copy
+  extend ActiveSupport::Concern
+
+  def copy(new_name = "Copy of " + name + Time.zone.now.to_s)
+    if template_valid? && type != 'ServiceTemplateAnsiblePlaybook'
+      ActiveRecord::Base.transaction do
+        dup.tap do |template|
+          template.update_attributes(:name => new_name, :guid => nil)
+          service_resources.each do |sr|
+            template.add_resource(sr.resource_type.constantize.find(sr.resource_id).try(:service_template_resource_copy))
```

Comment:
`sr.resource_type.constantize.find(sr.resource_id)` can be `sr.resource`, right?